egi2bids/mff2bids.py

```diff
@@ -47,80 +57,142 @@
 # fmt: on
 
 
+def _extract_folder(
+    file: Union[str, Path], dir_: Union[str, Path] = None
+) -> Path:
+    """Extract a .mff compressed folder to its original form."""
+    # check paths and file extension
+    file = _ensure_path(dir_, must_exist=True)
+    ext = file.suffix
+    _check_value(ext, (".tar", ".zip", ".mff"), "extension")
+    dir_ = Path.cwd() if dir_ is None else dir_
+    dir_ = _ensure_path(dir_, must_exist=True)
+    # open the archive if needed
+    archive_readers = {
+        ".tar": tarfile.open,
+        ".zip": zipfile.ZipFile,
+    }
+    if ext in (".tar", ".zip"):
+        logger.info("Extracting '%s' archive %s to %s.", ext, file, dir_)
+        with archive_readers[ext](file, "r") as archive:
+            archive.extractall(dir_)
+        for root, dirs, _ in os.walk(dir):
+            if "Contents" in dirs:
+                logger.info("MFF file found in %s", root)
+                return Path(root)
+        else:
+            raise (f"The '{ext}' archive does not contain a 'Content' folder.")
+
+    elif ext == ".mff":
+        return file
+
+
+@verbose
 def mff2bids(
-    mff_source,
-    bids_root,
-    subject=None,
-    task=None,
-    session=None,
+    mff_source: Union[str, Path],
+    bids_root: Union[str, Path],
+    subject,
+    session,
+    task,
     run=None,
     event_id=None,
-    line_frequency=50,
+    save_source: bool = False,
+    working_dir=None,
+    *,
+    overwrite: bool = False,
+    verbose=None,
 ):
```

Comment:
I removed `line_frequency` as this is a CB-specific application, and is definitely not going to be used in the US ;)
Couple of questions on the arguments. The only one used in the CLI are:

mff_source, bids_root, subject, task, session, run, save_source, overwrite and verbose.

What is `working_dir` here for?